src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/refreshanalyzer/RefreshSearchAnalyzerResponse.kt

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.indexmanagement.refreshanalyzer
+
+import org.apache.logging.log4j.LogManager
+import org.elasticsearch.action.support.DefaultShardOperationFailedException
+import org.elasticsearch.action.support.broadcast.BroadcastResponse
+import org.elasticsearch.common.io.stream.StreamInput
+import org.elasticsearch.common.io.stream.StreamOutput
+import org.elasticsearch.common.xcontent.ConstructingObjectParser
+import org.elasticsearch.common.xcontent.ToXContent.Params
+import org.elasticsearch.common.xcontent.XContentBuilder
+import org.elasticsearch.rest.action.RestActions
+import java.io.IOException
+import java.util.function.Function
+
+class RefreshSearchAnalyzerResponse : BroadcastResponse {
+
+    private var results: MutableMap<String, List<String>> = HashMap()
+    private var shardFailures: MutableList<FailedShardDetails> = mutableListOf()
+    private var temp: List<DefaultShardOperationFailedException> = mutableListOf()
+
+    protected var logger = LogManager.getLogger(javaClass)
+
+    @Throws(IOException::class)
+    constructor(inp: StreamInput) : super(inp) {
+        val resultSize: Int = inp.readVInt()
+        for (i in 0..resultSize) {
+            results.put(inp.readString(), inp.readStringArray().toList())
+        }
+
+        val failureSize: Int = inp.readVInt()
+        for (i in 0..failureSize) {
+            shardFailures.add(FailedShardDetails(inp.readString(), inp.readInt(), inp.readString()))
+        }
+    }
+
+    constructor(
+        totalShards: Int,
+        successfulShards: Int,
+        failedShards: Int,
+        shardFailures: List<DefaultShardOperationFailedException>,
+        results: MutableMap<String, List<String>>
+    ) : super(
+            totalShards, successfulShards, failedShards, shardFailures
+    ) {
+        this.results = results
+        this.temp = shardFailures
+        for (failure in shardFailures) {
```

Comment:
Can replace this with a `map`

Ex.
```
this.shardFailures = shardFailures.map{ FailedShardDetails(it.index()!!, it.shardId(), it.reason()) }
```

Reply:
I did some refactoring, this won't be needed now.